modules/host/network.go

```diff
@@ -174,6 +161,20 @@ func (h *Host) threadedListen() {
 	}
 	defer h.tg.Done()
 
+	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return
+	}
+	go func() {
+		defer h.tg.Done()
+		<-h.tg.StopChan()
+		err = h.listener.Close()
+		if err != nil {
+			h.log.Println("Closing the listener failed:", err)
+		}
+	}()
+
```

Comment:
1. The listener is started.
2. This thread is spawned, but no code is executed.
3. Close is called, the thread does not block because `Add` has not been called yet.